src/test/java/io/strimzi/kafka/bridge/http/ProducerIT.java

```diff
@@ -201,6 +201,103 @@ void sendSimpleMessageWithKey(VertxTestContext context) throws InterruptedExcept
         });
     }
 
+    @Test
+    void sendSimpleMessangeWithArrayKey(VertxTestContext context) throws InterruptedException, ExecutionException {
+        KafkaFuture<Void> future = adminClientFacade.createTopic(topic, 2, 1);
+
+        String value = "message-value";
+        JsonArray key = new JsonArray();
+        key.add("some-element").add(new JsonObject().put("some-field", "element-2"));
+
+        JsonArray records = new JsonArray();
+        JsonObject json = new JsonObject();
+        json.put("value", value);
+        json.put("key", key);
+        records.add(json);
+
+        JsonObject root = new JsonObject();
+        root.put("records", records);
+
+        future.get();
+
+        producerService()
+            .sendRecordsRequest(topic, root, BridgeContentType.KAFKA_JSON_JSON)
+            .sendJsonObject(root, verifyOK(context));
+
+        Properties consumerProperties = Consumer.fillDefaultProperties();
+        consumerProperties.setProperty(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, kafkaUri);
+
+        KafkaConsumer<byte[], String> consumer = KafkaConsumer.create(vertx, consumerProperties,
+                new ByteArrayDeserializer(), new KafkaJsonDeserializer<>(String.class));
+        consumer.handler(record -> {
+            context.verify(() -> {
+                assertThat(record.value(), is(value));
+                assertThat(record.topic(), is(topic));
+                assertThat(record.partition(), notNullValue());
+                assertThat(record.offset(), is(0L));
+                assertThat(record.key(), is(key.toBuffer().getBytes()));
```

Comment:
@ppatierno I am not sure if this is the best way to assert the key(which is in bytes). 
Any suggestions?

Reply:
The mistake here is totally different. Why did you delete the headers check? Kafka headers (with key-value) and key in the message are two totally different things. You should revert back the headers check part wherever you removed it.

Reply:
> The mistake here is totally different. Why did you delete the headers check? Kafka headers (with key-value) and key in the message are two totally different things. You should revert back the headers check part wherever you removed it.

Hmmm, I thought it would be better to follow other tests conventions. 
For example,  the `sendSimpleMessageWithKey(VertxTestContext context)` does not do any `header` check.

Reply:
Yeah ok makes sense, even because we are not sending headers here as well.